sklearn/kernel_ridge/kernel_ridge.py

```diff
@@ -0,0 +1,165 @@
+"""Module :mod:`sklearn.kernel_ridge` implements kernel ridge regression."""
+
+# Authors: Mathieu Blondel <mathieu@mblondel.org>
+#          Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+#          Carlos Perales <sir.perales@gmail.com>
+# License: BSD 3 clause
+
+import numpy as np
+
+from ..base import BaseEstimator, RegressorMixin
+from ..metrics.pairwise import pairwise_kernels
+from ..linear_model.ridge import _solve_cholesky_kernel
+from ..utils import check_array, check_X_y
+from ..utils.validation import check_is_fitted
+from ..utils.multiclass import unique_labels
+
+
+class KernelRidge(BaseEstimator, RegressorMixin):
+    """Kernel ridge regression and Kernel ridge classification
+    (also known as Kernel Extreme Learning Machine).
+    """
+    # By default, it works as a regressor
+    regression = True
+
+    def __init__(self, alpha=1, kernel="rbf", gamma=None, degree=3, coef0=1,
+                 kernel_params=None):
+        self.alpha = alpha
+        self.kernel = kernel
+        self.gamma = gamma
+        self.degree = degree
+        self.coef0 = coef0
+        self.kernel_params = kernel_params
+
+    def _get_kernel(self, X, Y=None):
+        if callable(self.kernel):
+            params = self.kernel_params or {}
+        else:
+            params = {"gamma": self.gamma,
+                      "degree": self.degree,
+                      "coef0": self.coef0}
+        return pairwise_kernels(X, Y, metric=self.kernel,
+                                filter_params=True, **params)
+
+    @property
+    def _pairwise(self):
+        return self.kernel == "precomputed"
+
+    def fit(self, X, y=None, sample_weight=None):
```

Comment:
not a lot of fit seems to be shared between the regressor and classifier. why not just put each fit into its corresponding class.